presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -184,12 +187,14 @@ public synchronized void addChangeListener(MemoryPoolId poolId, Consumer<MemoryP
         changeListeners.computeIfAbsent(poolId, id -> new ArrayList<>()).add(listener);
     }
 
-    public synchronized void process(Iterable<QueryExecution> runningQueries)
+    public synchronized void process(Iterable<QueryExecution> runningQueries, Supplier<List<QueryInfo>> allQueryInfoSupplier)
     {
         if (!enabled) {
             return;
         }
 
+        memoryLeakDetector.checkForMemoryLeaks(allQueryInfoSupplier, pools.get(GENERAL_POOL).getQueryMemoryReservations());
```

Comment:
- add a comment explaining we're not concerned about RESERVED_POOL
- what about revocable memory reservations? Those could leak as well

Reply:
Also, `com.facebook.presto.memory.ClusterMemoryPool#getQueryMemoryReservations` looks very unsafe. It returns internal, `@GuardedBy` `HashMap` to the outside world.

Reply:
- Revocable memory reservations are out of scope for this PR. I will add a TODO.
- I will add a separate commit to fix the `ClusterMemoryPool#getQueryMemoryReservations` issue as it is not introduced/changed as part of this PR.

Reply:
There is also a `com.facebook.presto.memory.LocalMemoryManager#RESERVED_POOL`. Add something like
```
// RESERVED_POOL is not check to simplify the code, as reserved pool is a legacy mechanism
```